fails with ghc-9.4 for Stackage Nightly

```
uilding library for async-pool-0.9.1..                                                                                  [30/182284]
[1 of 3] Compiling Control.Concurrent.Async.Pool.Async                                                                              

/var/stackage/work/unpack-dir/unpacked/async-pool-0.9.1-525b46e6a39c80b7461e8cd146d540c27d47abc717b746ff34c21aa9b1c47743/Control/Con
current/Async/Pool/Async.hs:714:16: error:
    • Couldn't match expected type: State# RealWorld                                                                                
                                    -> (# State# RealWorld, a1 #)                                                                   
                  with actual type: IO ()                         
    • In the first argument of ‘fork#’, namely ‘action’                                                                             
      In the expression: fork# action s                           
      In the expression:                                          
        case (fork# action s) of (# s1, tid #) -> (# s1, ThreadId tid #)                                                            
    |                                                             
714 |    case (fork# action s) of (# s1, tid #) -> (# s1, ThreadId tid #)                                                           
    |                ^^^^^^                                       

/var/stackage/work/unpack-dir/unpacked/async-pool-0.9.1-525b46e6a39c80b7461e8cd146d540c27d47abc717b746ff34c21aa9b1c47743/Control/Con
current/Async/Pool/Async.hs:719:22: error:
    • Couldn't match expected type: State# RealWorld                                                                                
                                    -> (# State# RealWorld, a0 #)                                                                   
                  with actual type: IO ()                         
    • In the second argument of ‘forkOn#’, namely ‘action’                                                                          
      In the expression: forkOn# cpu action s                                                                                       
      In the expression:                                          
        case (forkOn# cpu action s) of                            
          (# s1, tid #) -> (# s1, ThreadId tid #)                                                                                   
    |                                                             
719 |    case (forkOn# cpu action s) of (# s1, tid #) -> (# s1, ThreadId tid #)                                                     
    |                      ^^^^^^              
```

--- a/Control/Concurrent/Async/Pool/Async.hs
+++ b/Control/Concurrent/Async/Pool/Async.hs
@@ -723,10 +723,18 @@ tryAll = try
 -- exception handler.
 {-# INLINE rawForkIO #-}
 rawForkIO :: IO () -> IO ThreadId
+#if MIN_VERSION_base(4,17,0)
+rawForkIO (IO action) = IO $ \ s ->
+#else
 rawForkIO action = IO $ \ s ->
+#endif
    case (fork# action s) of (# s1, tid #) -> (# s1, ThreadId tid #)
 
 {-# INLINE rawForkOn #-}
 rawForkOn :: Int -> IO () -> IO ThreadId
+#if MIN_VERSION_base(4,17,0)
+rawForkOn (I# cpu) (IO action) = IO $ \ s ->
+#else
 rawForkOn (I# cpu) action = IO $ \ s ->
+#endif
    case (forkOn# cpu action s) of (# s1, tid #) -> (# s1, ThreadId tid #)

